### PR TITLE
boards: meshme  CC2538 + CC2592 initial implementation

### PR DESCRIPTION
**Pull request by luisan00**
### Contribution description
**meshme** is in a early stage, it combine the CC2538 SoC and a  CC2592 Front End, both from Texas Instruments.
This PR, add support for the "meshme" board, at least some basic support 

### Testing procedure
build and flash the board with some networking example

### Issues/PRs references
None

### CODE REVIEW

File: boards/meshme/Kconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    default y
+    select CPU_MODEL_CC2538SF53
+    select HAS_PERIPH_ADC
+    select HAS_PERIPH_I2C
+    select HAS_PERIPH_RTT
+    select HAS_PERIPH_SPI
+    select HAS_PERIPH_TIMER
+    select HAS_PERIPH_UART
+    select HAS_RIOTBOOT
+
```
Comment (eduazocar) [inline boards/meshme/Kconfig:21]:
```suggestion

```

File: boards/meshme/Makefile.dep [hunk 2]
```diff
@@ -0,0 +1,7 @@
+ifneq (,$(filter netdev_default,$(USEMODULE)))
+  USEMODULE += cc2538_rf
+endif
+
+ifneq (,$(filter saul_default,$(USEMODULE)))
+  USEMODULE += saul_gpio
+endif
```
Comment (eduazocar) [inline boards/meshme/Makefile.dep:7]:
```suggestion
endif

```

File: boards/meshme/Makefile.include [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# the Board serial numbers all begin with "06EB":
+PROGRAMMER_SERIAL ?= 06EB
+
+# setup serial terminal
+# the debug UART is always the second tty with the matching serial number:
+PORT_LINUX ?= $(word 2,$(shell $(RIOTTOOLS)/usb-serial/find-tty.sh '^$(PROGRAMMER_SERIAL)'))
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbserial*)))
+
+include $(RIOTBOARD)/common/cc2538/Makefile.include
```
Comment (eduazocar) [inline boards/meshme/Makefile.include:9]:
```suggestion
include $(RIOTBOARD)/common/cc2538/Makefile.include

```

File: boards/meshme/doc.txt [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/**
+@defgroup    boards_meshme-24g20p
+@ingroup     boards
+@brief       Mesh4All 2.4 GHz board + 20dBm PA
+
+# Overview
+MeshMe-24g20p is a complete communications system, it uses the CC2538 SoC MCU and
```
Comment (eduazocar) [inline boards/meshme/doc.txt:7]:
```suggestion
MeshMe-24g20p is a complete communications system, it uses the CC2538 SoC MCU and
```

**Review by eduazocar — CHANGES_REQUESTED** [review]
There are some checks found by the static-test, I propose the following suggestions

File: boards/meshme/Makefile.dep [hunk 5]
```diff
@@ -0,0 +1,8 @@
+ifneq (,$(filter netdev_default,$(USEMODULE)))
+  USEMODULE += cc2538_rf
+  USEMODULE += cc2538_rf_obs_sig
+endif
+
+ifneq (,$(filter saul_default,$(USEMODULE)))
+  USEMODULE += saul_gpio
+endif
```
Comment (eduazocar) [inline boards/meshme/Makefile.dep:8]: EOF

File: boards/meshme/include/periph_conf.h [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+#ifdef MODULE_PERIPH_UART_HW_FC
+        .cts_pin = GPIO_UNDEF,
+        .rts_pin = GPIO_UNDEF,
+#endif
+    },
+};
+
+#define UART_0_ISR isr_uart0               /*!< interrupt function name mapping */
+#define UART_NUMOF ARRAY_SIZE(uart_config) /*!< Number of UARTs */
+/** @} */
```
Comment (eduazocar) [inline boards/meshme/include/periph_conf.h:55]: This is throwing a doxygen warning documentation

**Review by eduazocar — CHANGES_REQUESTED** [review]
Everything looks good, only have a little details to merge. Go ahead :+1:

File: boards/meshme/include/periph_conf.h (hunk 6 above)
Comment (luisan00) [reply boards/meshme/include/periph_conf.h]: done

File: boards/meshme/include/gpio_params.h [hunk 7]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2017 DAI Labor Technische Universität Berlin
+ * Copyright (C) 2021 Mesh4all <contact@mesh4all.org>
```
[the remark this answers is not in the document]
Comment (luisan00) [reply boards/meshme/include/gpio_params.h]: done

File: boards/meshme/Makefile.dep (hunk 5 above)
Comment (luisan00) [reply boards/meshme/Makefile.dep]: done

**Comment by luisan00** [comment]
@RocioRojas @eduazocar  requested changes done! but  looks like some conflicts appear right now

**Comment by luisan00** [comment]
all changes done!

**Review by eduazocar — COMMENTED** [review]
I see all good, I think this is ready to merge, but first do the squash to approve it

File: boards/meshme/Makefile.dep (hunk 5 above)
[the remark this answers is not in the document]
Comment (luisan00) [reply boards/meshme/Makefile.dep]: outdated

**Comment by luisan00** [comment]
done @eduazocar !

**Comment by RocioRojas** [comment]
I looks good for me !!!

**Review by eduazocar — APPROVED** [review]
Good, it's time to merge it :+1:

**Pull request merged**